tap_impact/transform.py

```diff
@@ -47,12 +47,52 @@ def replace_order_id(this_json, data_key):
     return this_json
 
 
+# Replace system/reserved field 'oid' with 'order_id' in nested events
+# Also adjust events sub-node to always by a list/array (instead of array AND dict)
+def transform_conversion_paths(this_json, data_key):
+    i = 0
+    for record in this_json[data_key]:
+        events = record.get('events')
+        events_list = []
+        if isinstance(events, list):
+            events_list = events
+        elif isinstance(events, dict):
+            event = events.get('event')
+            events_list.append(event)
+        this_json[data_key][i].pop('events', None)
```

Comment:
Isn't this just `record.pop('events', None)`?

Reply:
It needs to be `this_json` because that is what I am returning.

Reply:
Python makes a lot of things mutable. So modifying `record` should modify the argument you passed to this function

```
>>> def modify_dict(my_dict):
...   for thing in my_dict['list']:
...     thing['a'] = 1000
...   return my_dict
... 
>>> dict_1 = {'list': [{'a': 0}, {'a': 1}, {'a': 2}]}
>>> 
>>> returned_dict = modify_dict(dict_1)
>>> returned_dict
{'list': [{'a': 1000}, {'a': 1000}, {'a': 1000}]}
>>> dict_1
{'list': [{'a': 1000}, {'a': 1000}, {'a': 1000}]}
```

Reply:
You can operate on the thing you are looping over. It gives you an error for that.
And I am returning `this_json` but do not want to return the events sub-node. That's why I am removing it with pop.